File: SecondYear/AvantikaSharma2307_AvantikaSharma_2226cseai1008_2/Week_2/armstrong.py
from os import *
from sys import *
from collections import *
from math import *
n=int(input())
a=n
b=n
sum1=0
count=0
while(n!=0):
    r=n%10
    count=count+1
    n=n//10
    
while(b!=0):
    r=b%10
    sum1=sum1+pow(r,count)
    b=b//10
if(sum1==a):
    print ("true")
else:
    print ("false")    

File: SecondYear/aayushraghav93_AayushRaghav_2226CSE(AI)1108_2/Week_2/Armstrong.java
import java.util.Scanner;

public class Armstrong {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int num = scanner.nextInt();
        
        if (isArmstrong(num)) {
			System.out.println(true);
           
        } else {
            System.out.println(false);
        }
    }

    public static boolean isArmstrong(int number) {
        int originalNumber, remainder, n = 0;
        double result = 0;

        originalNumber = number;

        // Count the number of digits
        int numDigits = countDigits(number);

        while (originalNumber != 0) {
            remainder = originalNumber % 10;
            result += Math.pow(remainder, numDigits);
            originalNumber /= 10;
        }

        // Check if the result is equal to the original number
        return (int) result == number;
    }

    public static int countDigits(int number) {
        int count = 0;
        while (number != 0) {
            number /= 10;
            count++;
        }
        return count;
    }
}
